scripts/cancel_github_workflows.py

```diff
@@ -0,0 +1,222 @@
+#!/usr/bin/env python3
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""
+Manually cancel previous GitHub Action workflow runs in queue.
+
+Example:
+  # Set up
+  export GITHUB_TOKEN=394ba3b48494ab8f930fbc93
+  export GITHUB_REPOSITORY=apache/incubator-superset
+
+  # cancel previous jobs for a PR
+  ./cancel_github_workflows.py 1042
+
+  # cancel previous jobs for a branch
+  ./cancel_github_workflows.py my-branch
+
+  # cancel all jobs
+  ./cancel_github_workflows.py 1024 --include-last
+"""
+import os
+from typing import Iterable, List, Optional, Union
+
+import click
+import requests
```

Comment:
Whenever possible, we should probably try to use `urllib3` over `requests` to avoid licensing issues.

Reply:
Isn't it using Apache 2.0 License as well?

Reply:
Oh wow, they've apparently relicensed. There was a refactor effort some time ago to remove explicit dependency on `requests` as it wasn't Apache compatible back then (see #7643). Good to know.

Reply:
Hmm, I can't find any reference of it ever having been anything but Apache 2.0. Apparently I'm misremembering something or this has been a major misunderstanding.

Reply:
I couldn't let this one go.. 😄 I read old mailing list archives and turned up this: https://lists.apache.org/thread.html/49c4d03d3aadab5c2d1c20ccf1cb7ca02edbc94961e6429dbf6a082b%40%3Cdev.superset.apache.org%3E Turns out `requests` was pulling in `chardet` which is LGPL, which was the problem. Turns out `chardet` was dropped as of 09/2020: https://github.com/encode/httpx/pull/1269 🎉

Reply:
So it means it's OK to use requests again, I guess?

Reply:
Uhh, I went down a link in this issue without noticing that it was pointing to another project: https://github.com/psf/requests/issues/4848 Apparently requests still uses chardet, and there are no plans to remove it. As long as this is the case we're not allowed to reference requests as a required dependency, but it's ok to be had as an optional dependency. As long as we restrict the usage to development scripts we're fine, but we shouldn't use it in the core code.

Reply:
Got it. Thanks for the clarification and investigation!